cloud/google/clients/computeservice.go

```diff
@@ -64,3 +90,73 @@ func (c *ComputeService) InstancesInsert(project string, zone string, instance *
 func (c *ComputeService) ZoneOperationsGet(project string, zone string, operation string) (*compute.Operation, error) {
 	return c.service.ZoneOperations.Get(project, zone, operation).Do()
 }
+
+// A pass through wrapper for compute.Service.GlobalOperations.Get(...)
+func (c *ComputeService) GlobalOperationsGet(project string, operation string) (*compute.Operation, error) {
+	return c.service.GlobalOperations.Get(project, operation).Do()
+}
+
+// A pass through wrapper for compute.Service.Firewalls.List(...)
+func (c *ComputeService) FirewallsGet(project string) (*compute.FirewallList, error) {
+	return c.service.Firewalls.List(project).Do()
+}
+
+// A pass through wrapper for compute.Service.Firewalls.Insert(...)
+func (c *ComputeService) FirewallsInsert(project string, firewallRule *compute.Firewall) (*compute.Operation, error) {
+	return c.service.Firewalls.Insert(project, firewallRule).Do()
+}
+
+// A pass through wrapper for compute.Service.Firewalls.Delete(...)
+func (c *ComputeService) FirewallsDelete(project string, name string) (*compute.Operation, error) {
+	return c.service.Firewalls.Delete(project, name).Do()
+}
+
+func (c *ComputeService) WaitForOperation(config *gceconfigv1.GCEClusterProviderConfig, op *compute.Operation) error {
+	glog.Infof("Wait for %v %q...", op.OperationType, op.Name)
+	defer glog.Infof("Finish wait for %v %q...", op.OperationType, op.Name)
+
+	start := time.Now()
+	ctx, cf := context.WithTimeout(context.Background(), gceTimeout)
+	defer cf()
+
+	var err error
+	for {
+		if err = c.checkOp(op, err); err != nil || op.Status == "DONE" {
+			return err
+		}
+		glog.V(1).Infof("Wait for %v %q: %v (%d%%): %v", op.OperationType, op.Name, op.Status, op.Progress, op.StatusMessage)
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("gce operation %v %q timed out after %v", op.OperationType, op.Name, time.Since(start))
+		case <-time.After(gceWaitSleep):
+		}
+		if op.Zone != "" {
+			op, err = c.getOp(config, op)
+		} else {
+			op, err = c.getGlobalOp(config, op)
+		}
+	}
+}
+
+// getOp returns an updated operation.
+func (c *ComputeService) getOp(config *gceconfigv1.GCEClusterProviderConfig, op *compute.Operation) (*compute.Operation, error) {
```

Comment:
maybe have it just take the project name as a string instead of passing down the config? as it doesn't seem to be really using anything else from the config.

Reply:
done. Also updated WaitForOperation to take the project instead of the config